File: 1570-final-prices-with-a-special-discount-in-a-shop/1570-final-prices-with-a-special-discount-in-a-shop.go
func finalPrices(prices []int) []int {
    n := len(prices)
    dis := make ([]int, 0)

    for i := 0; i < n - 1; i++ {
        d := prices[i]
        for j := i + 1; j < n; j++ {
            if prices[j] <= prices[i] {
                d = prices[i] - prices[j]
                break
            }
        }
        dis = append(dis, d)
    }
    dis = append(dis, prices[n - 1])
    return dis
}

File: 1570-final-prices-with-a-special-discount-in-a-shop/1570-final-prices-with-a-special-discount-in-a-shop.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* finalPrices(int* prices, int pricesSize, int* returnSize) {
    int n = pricesSize;
    int *dis = (int*)calloc(n, sizeof(int));
    *returnSize = 0;
    for(int i = 0; i < n - 1; ++i){
        int discount = prices[i];
        for(int j = i + 1; j < n ; ++j){
            if(prices[j] <= prices[i]){
                discount = prices[i] - prices[j];
                break;
            }
        }
        dis[(*returnSize)++] = discount;
    }
    dis[(*returnSize)++] = prices[n - 1];
    return dis;
}